### PR TITLE
Add Analytics section to API docs

### PR DESCRIPTION
**Pull request by KatieWright26**
This PR adds API information for our Flaky Test API. Some relevant points of discussion:
- https://linear.app/buildkite/issue/PIE-1291/onboarding-process-and-sharable-material-for-private-beta-users#comment-d80f1c81
- [Linear ticket](https://linear.app/buildkite/issue/PIE-1292/flaky-tests-public-api-documentation)

Screenshot 2023-02-21 at 16 46 33 (2)

### CODE REVIEW

**Comment by KatieWright26** [comment]
FYI @mbelton-buildkite for our catch up tomorrow

File: pages/apis/rest_api/analytics/flaky_tests.md.erb
Comment (mbelton-buildkite) [inline pages/apis/rest_api/analytics/flaky_tests.md.erb:5]: Let's style this as a warning because it will stop people from making progress if they're not aware they need to request access:

```suggestion
> 🚧 Beta feature
> Flaky test detection is in beta and not available to all organizations. To get early access, TBC.

{:toc}
```

A few nitpicks included in the suggestion:

- Be clear about what "this" refers to.
- Avoid time words like "currently" because they tend to lose meaning when published.
- Use "To X, do Y" structure for the instruction.
- Generally avoid "please" because it weakens the instruction, which is a required action here.

I also switched the order of the table of contents since the warning is important to know before you get into the content of the page.

File: pages/apis/rest_api/analytics/flaky_tests.md.erb
Comment (mbelton-buildkite) [inline pages/apis/rest_api/analytics/flaky_tests.md.erb:9]: This looks like it's tied to a test suite rather than a user? If so, perhaps:

```suggestion
Returns a [paginated list](<%= paginated_resource_docs_url %>) of the flaky tests detected in a test suite.
```

File: pages/apis/rest_api/analytics/flaky_tests.md.erb
Comment (mbelton-buildkite) [inline pages/apis/rest_api/analytics/flaky_tests.md.erb:12]: Is including the access token in a query parameter standard for Buildkite? I see in the overview that it only mentions using the `Authorization` header.

**Review by mbelton-buildkite — COMMENTED** [review]
It's looking good! I've left a few comments for you to consider 😊

File: pages/apis/rest_api/analytics/flaky_tests.md.erb
Comment (KatieWright26) [reply pages/apis/rest_api/analytics/flaky_tests.md.erb:12]: It can be added as a query param;  I'll remove it here though as it's not a pattern included in our other documentation

File: pages/apis/rest_api/analytics/flaky_tests.md.erb
Comment (mbelton-buildkite) [inline pages/apis/rest_api/analytics/flaky_tests.md.erb:7]: I think we can remove this now that the _Beta_ label is more apparent on the page.

File: pages/apis/rest_api/analytics/flaky_tests.md.erb
Comment (mbelton-buildkite) [inline pages/apis/rest_api/analytics/flaky_tests.md.erb:52]: Nitpick: Just include one newline at the end of files.

```suggestion
```

File: pages/apis/rest_api/analytics/flaky_tests.md.erb
Comment (mbelton-buildkite) [inline pages/apis/rest_api/analytics/flaky_tests.md.erb:2]: Nitpick: Separate headings and paragraphs with a blank line.

```suggestion
# Flaky tests API

The Flaky test API endpoint provides information about tests detected as flaky in a test suite.
```

**Review by mbelton-buildkite — APPROVED** [review]
Just a couple of nitpicks, but otherwise looks good 😊

**Comment by KatieWright26** [comment]
Hi @dannymidnight  - your commits have erased content from flaky_tests.md. Are you able to fix this?

Screenshot 2023-02-21 at 14 22 36 (2)

File: pages/apis/rest_api/analytics/flaky_tests.md
Comment (gchan) [inline pages/apis/rest_api/analytics/flaky_tests.md:28]: There is currently no optional query parameters to filter the results.

moot point - it seems the pattern in other pages is that the 'optional query string parameters' section comes after the response example. But if we intend to remove this section, this doesn't matter.

File: pages/apis/rest_api/analytics/flaky_tests.md
Comment (gchan) [inline pages/apis/rest_api/analytics/flaky_tests.md:37]: Maybe we could provide a real-ish example?

```suggestion
    "scope": "User#email",
    "name": "is correctly formatted",
    "location": "./spec/models/user_spec.rb:42",
    "file_name": "./spec/models/user_spec.rb",
```

Or have fun with a humorous or pop-culture reference

**Review by gchan — COMMENTED** [review]
I took a look at the latest preview [here](https://bk-docs-pr-1877.onrender.com/docs/apis/rest-api/analytics/flaky-tests) and it looks good. It matches the format of the other API pages! 

I had a question about the 'optional query string parameter' section.

File: pages/apis/rest_api/analytics/flaky_tests.md
Comment (KatieWright26) [reply pages/apis/rest_api/analytics/flaky_tests.md]: Oh weird, this optional endpoint was mentioned in another PR. Have removed.

File: pages/apis/rest_api/analytics/flaky_tests.md
Comment (gchan) [inline pages/apis/rest_api/analytics/flaky_tests.md:19]: Perhaps use a random UUID?

```suggestion
    "id": "01867216-8478-7fde-a55a-0300f88bb49b",
    "web_url": "/organizations/my_great_org/analytics/suites/my_suite_name/tests/01867216-8478-7fde-a55a-0300f88bb49b",
```

**Review by gchan — APPROVED** [review]
LGTM! Do we have to wait till the public-beta launch before merging like the other documentation PR? Or can we ship it right now? This might be a decision for product and/or marketing.

**Comment by KatieWright26** [comment]
Holding off merge while waiting on go-ahead from production/marketing @meiqimichelle

**Pull request merged**
